backend/core/ai/graph_nodes.py

```
@@ -0,0 +1,380 @@
+        return {"history_id": saved.id}
+    
+    def detect_anomaly_node(self, state: AgentState) -> Dict[str, Any]:
+        """노드 6: 이상 감지 (단일 값만 지원)"""
+        print(f"\n[노드 6] 이상 감지 중...")
+        
+        metric_id = state.get('metric_id')
+        history_id = state.get('history_id')
+        threshold_config = state.get('threshold_config')
+        
+        if not all([metric_id, history_id, threshold_config]):
+            print(f"  ⚠ 이상 감지 스킵 (설정 없음)")
```

Ruff (0.13.1)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)